include/exadg/grid/grid_utilities.h

```diff
@@ -245,6 +246,52 @@ create_fine_and_coarse_triangulations(
   }
 }
 
+/**
+ * This function reads an external triangulation. The path to the triangulation file should
+ * be defined in the input file and then stored in GridData. This function extracts it from
+ * GridData and stores it in the triangulation "tria".
+ */
+template<int dim>
+inline void
+read_external_triangulation(dealii::Triangulation<dim, dim> & tria, GridData const & data)
+{
+  AssertThrow(!data.file_name.empty(),
+              dealii::ExcMessage(
+                "You are trying to read a grid file, but the string, which is supposed to contain"
+                " the file name, is empty. Most likely, you forgot to specify the file location in"
+                " the input file."));
+
+  dealii::GridIn<dim> grid_in;
+
+  grid_in.attach_triangulation(tria);
+
+  // find the file extension from the file name
+  std::string extension = data.file_name.substr(data.file_name.find_last_of('.') + 1);
+
+  AssertThrow(!extension.empty(),
+              dealii::ExcMessage("You are trying to read a grid file, but the file extension is"
+                                 " empty. Check the input file to make sure that the file"
+                                 " extension is correctly defined after a full stop."));
```

Comment:
Another reference to input file. I might be that this function is not properly designed in terms of the single-responsibility principle. Please rethink how to design this function: which parameters are passed to this function? `GridData` contains much more information than what is needed here.

Reply:
I decided to pass `GridData` to this function because:

1. The user is forced to use `GridData`, which is in my opinion good. I could also go with a `std::string` argument, but then everyone could define the path in application directly. (Not that they can't do it now, but it is harder and therefore less likely.)
2. I can check the mismatch of element types bellow with an Assert.

Do you think that a `std::string` argument would be better?

Reply:
They are the same, that is why I said : 

> (Not that they can't do it now, but it is harder and therefore less likely.)

For now, I only need the `file_name` and `element_type`. The `element_type` is only needed for the Assert. I have checked the variables of `GridData` again, and it seems unlikely that this function would end up needing more variables. 

I will change to a `std::string` argument and omit the Assert for now. But I think the Assert is needed somewhere.

Reply:
Let's keep this in mind for the future.